internal/controller/reporter/reporter.go

```diff
@@ -0,0 +1,40 @@
+package controller
+
+import (
+	"fmt"
+	"github.com/superhacker2002/parentheses/internal/parentheses"
+)
+
+type client interface {
+	Generate(length uint) (string, error)
+}
+
+type Reporter struct {
+	Client client
```

Comment:
If you'll have New() contruct function client could be private filed, so noone of of package could use you client